### ISSUE TITLE
Usernames are case-sensitive when mentioning someone

### ISSUE THREAD

**Issue by fleaz** [RocketChat/Rocket.Chat#13374]
### Description:

When creating users in RocketChat they are treated case insensitive, e.g. I can't create two users with the username "foo" and "Foo". But apparently when mentioning someone in a channel, it only gets converted to a highlight if one uses the correct case. This bites a lot of our customers because to to pretty short names some of them just type the message including the highlight instead of using the autocompletion dropdown.

<img width="225" alt="screenshot 2019-02-05 at 17 38 52" src="https://user-images.githubusercontent.com/2489598/52288686-f4b1a380-296c-11e9-9241-81bdab7cd992.png">

### Steps to reproduce:

1. Create a user called "foo"
2. Go to a channel and mention "@Foo"

### Expected behavior:

The user "foo" gets a mention because usernames are case-insensitive in the rest of the system

### Actual behavior:

No mention is created.

### Server Setup Information:

- Version of Rocket.Chat Server: 0.74.0
- Operating System: Linux
- Deployment Method: Hosted Cloud Version
- NodeJS Version: v8.11.4

**Comment by thayannevls** [comment]
Is anyone working on this? Can I try to do the mentions case-insensitive?

**Comment by Rodriq** [comment]
Gretings @fleaz if I got you correctly,
  we can not create a user **``Example``** and another **``example``**.
 if say user **`Example`** truly exist 
      then when we mention  **`example`** we don't get a tag. Whereas if mention **`Example`** we get a tag??

**Comment by fleaz** [comment]
> Gretings @fleaz if I got you correctly,
> we can not create a user **`Example`** and another **`example`**.
> if say user **`Example`** truly exist
> then when we mention **`example`** we don't get a tag. Whereas if mention **`Example`** we get a tag??

This is correct, yes :)

If you take a look at my screenshot the two users `felix` and `TestKunde` exist. But the mentioning of `@Felix` oder `@testkunde` didn't create a highlight

**Comment by reetp** [comment]
Can you test the search on https://open.rocket.chat/ as it seems to be case insensitive there?

I can search on both Reetp and reetp and get the same result? Both get highlighted in the message?

**Comment by Rodriq** [comment]
> Can you test the search on https://open.rocket.chat/ as it seems to be case insensitive there?
> 
> I can search on both Reetp and reetp and get the same result? Both get highlighted in the message?

Yes @reetp I found out so this is what I got by trying on the #sandbox pane

![case sensitive](https://user-images.githubusercontent.com/39289592/52571978-008ee100-2e17-11e9-9e23-8b8dee9e89f2.png)

**Comment by fleaz** [comment]
As stated in my initial description of this issue,  it only occures when you type the username by hand and **not** use the autocompletion popup which appers while you start typing.

Especially for short names (e.g. tom) it's faster to just keep writing instead of dealing with the popup and selecting the right user

**Comment by algomaster99** [comment]
I am working on this. Will submit a PR soon :)

**Comment by algomaster99** [comment]
Can I get some help here on how to go about it? I have understood the flow but can't test it properly. I can't run `console.log()` in `Rocket.Chat/packages/rocketchat-mentions/server/Mentions.js`. The fix will be there only I suppose.

**Comment by fleaz** [comment]
Can anybody help @algomaster99 with this?

Our customer are running into this issue regularly :(

**Comment by ashwaniYDV** [comment]
I am working on this

**Comment by ashwaniYDV** [comment]
I am working on it.

**Comment by ashwaniYDV** [comment]
@fleaz I have done the changes. Please review my PR #16487 :)

**Comment by fleaz** [comment]
Hey, any updates on this? This is super annoying and now over a year old :(

**Comment by skunapuli03** [comment]
Hey, any updates on this?
